### PR TITLE
Fix equals bugs which do not work due to the wrong classes being compared

### PR DESCRIPTION
**Pull request by GameCubeGBA**

### CODE REVIEW

File: Ghidra/Debug/Debugger-agent-frida/src/main/java/agent/frida/manager/cmd/AbstractFridaCommand.java [hunk 1]
```diff
@@ -117,8 +115,9 @@ public void parse(String result, Object data) {
 			} else {
 				manager.getEventListeners().fire.consoleOutput(object+"\n", 0);		
 			}
-			if (res.equals("[]")) {
+			if (res.toString().equals("[]")) {
```
Comment (dev747368) [inline Ghidra/Debug/Debugger-agent-frida/src/main/java/agent/frida/manager/cmd/AbstractFridaCommand.java:118]: would be better if you flipped the const and non-const around: "[]".equals(res.toString())

File: Ghidra/Features/Base/src/main/java/ghidra/app/plugin/core/datamgr/archive/FileArchive.java [hunk 2]
```diff
@@ -283,7 +283,7 @@ public void saveAs(Component component) throws IOException {
 			return;
 		}
 
-		if (saveAsFile.equals(getFile())) {
+		if (saveAsFile.getName().equals(getFile().getName())) {
```
Comment (dev747368) [inline Ghidra/Features/Base/src/main/java/ghidra/app/plugin/core/datamgr/archive/FileArchive.java:286]: this needs to change to compare the full file path, not just the name.

Suggest ```saveAsFile.equals(archiveFile.getFile(false))```

File: Ghidra/Features/Base/src/main/java/ghidra/app/plugin/core/programtree/DragNDropTree.java [hunk 3]
```diff
@@ -195,7 +195,7 @@ public boolean isDropOk(DropTargetDragEvent e) {
 			}
 
 			// This is tree node transferable... 
-			if (draggedNodes.equals(root)) {
+			if (draggedNodes[0].equals(root)) {
```
Comment (dev747368) [inline Ghidra/Features/Base/src/main/java/ghidra/app/plugin/core/programtree/DragNDropTree.java:198]: would need to check that draggedNodes.length > 0 before indexing the array

File: Ghidra/Features/Base/src/main/java/ghidra/app/plugin/core/references/EditExternalReferencePanel.java [hunk 4]
```diff
@@ -160,11 +160,8 @@ private void populateExternalNames() {
 		extLibName.clearModel();
 		extLibName.addItem(Library.UNKNOWN);
 		Arrays.sort(names);
-		for (int i = 0; i < names.length; i++) {
-			if (Library.UNKNOWN.equals(extLibName)) {
-				continue;
-			}
-			extLibName.addItem(names[i]);
+		for (String name : names) {
```
Comment (dev747368) [inline Ghidra/Features/Base/src/main/java/ghidra/app/plugin/core/references/EditExternalReferencePanel.java:163]: you omitted the (admittedly faulty) logic to skip the Library.UNKNOWN name.

**Comment by GameCubeGBA** [comment]
Fixed!

File: Ghidra/Features/Base/src/main/java/ghidra/app/plugin/core/references/EditExternalReferencePanel.java [hunk 5]
```diff
@@ -160,11 +160,10 @@ private void populateExternalNames() {
 		extLibName.clearModel();
 		extLibName.addItem(Library.UNKNOWN);
 		Arrays.sort(names);
-		for (int i = 0; i < names.length; i++) {
-			if (Library.UNKNOWN.equals(extLibName)) {
-				continue;
+		if (!Library.UNKNOWN.equals(extLibName)) {
```
Comment (dev747368) [inline Ghidra/Features/Base/src/main/java/ghidra/app/plugin/core/references/EditExternalReferencePanel.java:163]: this still isn't correct.  The original code was trying to compare Library.UNKNOWN (ie. ```"<EXTERNAL>"```) to the current element from names.

```java
		for (String name : names) {
			if (Library.UNKNOWN.equals(name)) {
				continue;
			}
			extLibName.addItem(name);
		}
```

Comment (GameCubeGBA) [reply Ghidra/Features/Base/src/main/java/ghidra/app/plugin/core/references/EditExternalReferencePanel.java:163]: That was not what the original code did. At all.

Comment (GameCubeGBA) [reply Ghidra/Features/Base/src/main/java/ghidra/app/plugin/core/references/EditExternalReferencePanel.java:163]: Fixed though

**Comment by GameCubeGBA** [comment]
Fixed!

File: Ghidra/Features/Base/src/main/java/ghidra/app/plugin/core/references/EditExternalReferencePanel.java (hunk 5 above)
Comment (dev747368) [reply Ghidra/Features/Base/src/main/java/ghidra/app/plugin/core/references/EditExternalReferencePanel.java]: Yeah, it wasn't doing it, but I could tell that was what it was trying to do. :crossed_fingers:

**Pull request merged**